src/core/client-request.model.ts

```diff
@@ -0,0 +1,3 @@
+export type QueryParams = {
+    [k: string]: string | number | boolean
```

Comment:
We should have type completion for the known parameters if this now is part of the public API. 
Also we need test for each of the cases, as im not sure these params are supported everywhere